mirror-registry/README.md

```diff
@@ -1,106 +1,111 @@
+1. Gather the binary at [https://developers.redhat.com/content-gateway/file/pub/openshift-v4/clients/mirror-registry/1.2.3/mirror-registry.tar.gz](https://developers.redhat.com/content-gateway/file/pub/openshift-v4/clients/mirror-registry/1.2.3/mirror-registry.tar.gz) and untar it:
 
-```
-curl -s -L https://developers.redhat.com/content-gateway/file/pub/openshift-v4/clients/mirror-registry/1.0/mirror-registry.tar.gz | tar xvz -C /usr/bin
-```
+    ~~~sh
+    curl -s -L https://developers.redhat.com/content-gateway/file/pub/openshift-v4/clients/mirror-registry/1.2.3/mirror-registry.tar.gz | tar xvz -C /usr/bin
+    ~~~
+
+2. Use it in a simple way:
+
+    > :information_source: This will use an auto generated cert stored in /etc/quay-install and create a random password for the `init` user. You will have to use Podman with insecure mode or trust said certificate in order to interact with the resulting registry.
+
+    ~~~sh
+    /usr/bin/mirror-registry install
+    ~~~
+
+3. You can also be more specific:
+
+    * `REGISTRY_NAME` is the fqdn of your box
+    * `SSL_CERT` is the path of a valid CA (as generated by openssl)
+    * `SSL_KEY` is the path of the corresponding ssl key
 
-2. Use it in a simple way
+    1. Generate the self-signed certificates:
 
-```
-/usr/bin/mirror-registry install
-```
+        ~~~sh
+        REGISTRY_NAME=$(hostname -f)
+        sudo mkdir -p /etc/quay-install/certs/
+        sudo chown -R $(id -u):$(id -g) /etc/quay-install
 
-This will use an autogenerated cert stored in /etc/quay-install and create a random password for the init user. You will have to use podman with insecure mode or trust said certificate in order to interact with the resulting registry
+        openssl req -x509 -newkey rsa:4096 -keyout /etc/quay-install/certs/registry-key.pem -nodes -out /etc/quay-install/certs/registry-cert.pem -sha256 -days 9999 -subj "/CN=${REGISTRY_NAME}" -addext "subjectAltName = DNS:${REGISTRY_NAME}"
+        ~~~
 
-3. You can also be more specific
+    2. Run the registry install:
 
-- REGISTRY_NAME is the fqdn of your box
-- SSL_CERT is the path of a valid CA (as generated by openssl)
-- SSL_KEY is the path of the corresponding ssl key
+        > :information_source: The command below will initialize the registry with the user `admin` and the password `n0tTh4tS3cur3`.
 
-```
-SSH_KEY="/root/.ssh/id_rsa"
-/usr/bin/mirror-registry install --quayHostname ${REGISTRY_NAME} --sslCert ${SSL_CERT} --sslKey ${SSL_KEY} --initPassword ${REGISTRY_PASSWORD} --ssh-key ${SSH_KEY}
-```
+        ~~~sh
+        SSH_KEY="${HOME}/.ssh/id_rsa"
+        /usr/bin/mirror-registry install --quayHostname ${REGISTRY_NAME} --sslCert /etc/quay-install/certs/registry-cert.pem --sslKey /etc/quay-install/certs/registry-key.pem --initUser admin --initPassword n0tTh4tS3cur3 --ssh-key ${SSH_KEY}
```

Comment:
i think at this point, we can specify a username too?

Reply:
It was there `--initUser`.